src/Model/Item.php

```diff
@@ -2861,17 +2878,16 @@ private static function addLinkAttachment(array $attachments, array $item, strin
 					$data['preview'] = $attachment['preview'] ?? '';
 				}
 			}
-		} elseif (preg_match("/.*(\[attachment.*?\].*?\[\/attachment\]).*/ism", $item['body'], $match)) {
-			$footer = $match[1];
-			$data = [];
+		} elseif (preg_match("/.*(\[attachment.*?\].*?\[\/attachment\]).*/ism", $body, $match)) {
+			$data = BBCode::getAttachmentData($match[1]);
 		}
 		DI::profiler()->saveTimestamp($stamp1, 'rendering');
 
-		if (!empty($footer) || !empty($data)) {
+		if (!empty($data) && !in_array($data['url'], $ignore_links)) {
```

Comment:
Since this is the only key we are using in the `$data` array, we should check for its specific existence.
```suggestion
		if (isset($data['url']) && !in_array($data['url'], $ignore_links)) {
```